src/v/cluster/health_monitor_backend.cc

```diff
@@ -271,6 +277,271 @@ void health_monitor_backend::abort_current_refresh() {
+            continue;
+        }
+        for (const auto& [nt, partitions] : node_report->topics) {
+            for (const auto& partition_status : partitions) {
+                auto nt_it = unclaimed_partitions.find(nt);
+                if (nt_it == unclaimed_partitions.end()) {
+                    continue;
+                }
+                auto p_it = nt_it->second.find(partition_status.id);
+                if (p_it == nt_it->second.end()) {
+                    continue;
+                }
+                // replica exists locally and followers_stats not processed yet
+                if (const auto& fs = partition_status.followers_stats) {
+                    vlog(
+                      clusterlog.trace,
+                      "leader found on node_id={} nt={} pid={} fs={}",
+                      node_id,
+                      nt,
+                      partition_status.id,
+                      *fs);
+                    remote_leader_handler(*fs, nt, partition_status.id);
+
+                    // to ignore other leaders, should there be more than one
+                    nt_it->second.erase(p_it);
+                    if (nt_it->second.empty()) {
+                        unclaimed_partitions.erase(nt_it);
+                    }
+                }
+            }
+        }
+    }
+
+    // Partitions eventually unclaimed, no node says it hosts their leaders.
+    for (const auto& [nt, partitions] : unclaimed_partitions) {
+        for (auto pid : partitions) {
+            unclaimed_partition_handler(nt, pid);
+        }
+    }
+    return errc::success;
+}
+
+ss::future<result<restart_risk_report>>
+health_monitor_backend::get_current_node_restart_risks(
+  size_t limit, model::timeout_clock::time_point deadline) {
+    auto holder = _gate.hold();
+    if (
+      auto ec = co_await maybe_refresh_cluster_health(
+        force_refresh::no, deadline)) {
+        co_return ec;
+    }
+    if (!_restart_risks_collected) {
+        // technically it may be already enabled at the moment,
+        // but was disabled on last collection
+        co_return make_error_code(errc::feature_disabled);
+    }
+
+    restart_risk_report report{.limit = limit};
+    auto report_handler = [&report, this](
+                            const followers_stats& fs,
+                            const model::topic_namespace& nt,
+                            model::partition_id pid) {
+        record_risks_in_report(report, _self, fs, nt, pid);
+    };
+    auto ec = walk_local_and_remote_reports(
+      report_handler,
+      report_handler,
+      [&report,
+       this](const model::topic_namespace& nt, model::partition_id pid) {
+          auto rf = _topic_table.local().get_topic_replication_factor(nt);
+          if (rf == replication_factor{1}) {
+              report.push(&restart_risk_report::rf1_offline, nt, pid);
+          } else {
+              report.push(&restart_risk_report::acks1_data_loss, nt, pid);
```

Comment:
This is just a leaderless partition, why are we recording this risk here? I guess we cannot know for sure, all 3 replicas may have been in sync just before it lost leadership, then restarting a node is safe.

Reply:
Yes I'm reporting the worst risks as we cannot know for sure.